### PR TITLE
New rule: disallowSwitchCaseIndentation

### LINKED ISSUE #1111: Request: disallowSwitchCaseIndentation
If this flag is true, then `case` statements should live at the same indentation level as their parent `switch`, a'la Crockford style.

Valid:
````javascript
switch (x) {
case 'b':
    break;
}
````

Invalid:
````javascript
switch (x) {
    case 'b':
        break;
}
````

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8852174-request-disallowswitchcaseindentation?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by yaycmyk**
As is the requirement in Crockford style, this rule enforces that direct-child
case statements within a switch block must not be indented or have a different
starting column than their parent switch. This is enforced on a per-switch basis,
so a switch nested inside another could itself be indented with the former's cases
having to be at the same indentation level.

Closes gh-1111
Fixes #1111

### CODE REVIEW

**Comment by mikesherov** [comment]
@yaymck, thanks for contributing! This is better handled as an option to the `validateIndentation` rule. Care to give it a shot there instead?

That rule already allows both styles... should be easy enough to enforce based on a Config value. Otherwise, I can handle it myself. Let me know... Either way I'm going to close this for now.

**Comment by mikesherov** [comment]
@yaycmyk ^

**Comment by yaycmyk** [comment]
@mikesherov I tried playing around in that ruleset and found it pretty hard to follow - will let you grab this one if it absolutely can't live as its own rule.

**Comment by mikesherov** [comment]
Yes, it should not live as its own rule.
